custom_components/deltasol/deltasolapi.py

```diff
@@ -47,31 +46,35 @@ def fetch_data(self):
         """Use api to get data"""
 
         response = {}
-        
-        if self.api_mode == "km2":
-            url = "http://" + self.host + "/cgi-bin/resol-webservice"
+
+        try:
+            url = f"http://{self.host}/cgi-bin/resol-webservice"
 
             headers = {
               'Content-Type': 'application/json'
             }
 
             payload = "[{'id': '1','jsonrpc': '2.0','method': 'login','params': {'username': '" + self.username + "','password': '" + self.password + "'}}]"
             response = requests.request("POST", url, headers=headers, data = payload).json()
-            _LOGGER.debug(f"KM2 Login result {response}")
+
+            _LOGGER.debug("Got valid JSON, assuming that we have a KM2 device ...")
 
             authId = response[0]['result']['authId']
             
             payload = "[{'id': '1','jsonrpc': '2.0','method': 'dataGetCurrentData','params': {'authId': '" + authId + "'}}]"
-            
+
             response = requests.request("POST", url, headers=headers, data = payload).json()
             _LOGGER.debug(f"KM2 response {response}")
             response = response[0]["result"]
+
+        except:
```

Comment:
Perhaps check explicitly for `ValueError` ?
In that case we don't mask other exceptions that might occur for example when doing the post request.